Feature/enhancements to the data request functionality

MarkCalvert opened the pull request:
https://salsadigital.atlassian.net/browse/DQL2-30
https://salsadigital.atlassian.net/browse/DQL2-31
https://salsadigital.atlassian.net/browse/DQL2-32

Commit:
[DQL2-30] Updated new_datarequest_organisation email template

Commit:
[DQL2-31]
- Added 'close_circumstance' & 'approx_publishing_date' to 'close_datarequest' action
- On 'open_datarequest' set 'close_circumstance' & 'approx_publishing_date' to None
- Refactored '_dictize_datarequest' & '_undictize_datarequest_basic' methods to the 'db.py' model
- Removed unused code after the refactor

Commit:
[DQL2-31]
- Updated _get_organization to use model_dictize.group_dictize of organization model object

Commit:
Merge remote-tracking branch 'origin/develop' into feature/Enhancements_to_the_data_request_functionality

duttonw commented on ckanext/data_qld/actions.py:
```diff
@@ -7,6 +7,7 @@
 import ckanext.datarequests.validator as validator
 import datetime
 import logging
+import ckan.lib.dictization.model_dictize as model_dictize
```
good to see library reuse 👍

duttonw commented:
can you enable circleci builds on your repo's (unsure if this is already done)

Commit:
[DQL2-31]
- Updated update_datarequest action to refresh datarequest object from DB to retrieve updated organisation relationship object

Commit:
[DQL2-31]
- Reverted changes back to original way of using '_dictize_datarequest' & '_undictize_datarequest_basic' and not using model relationships

Commit:
[DQL2-31]
Updated 'config.feature' as the "Data Request Config" has been removed

duttonw commented on test/features/config.feature:
```diff
@@ -5,6 +5,5 @@ Feature: Config
         Given "Admin" as the persona
         When I log in
         And I visit "ckan-admin/config"
-        Then I should see "Data Request Config"
```
cucumber is given when then, with and's.  Please create a then answer for this test so the logic still works

duttonw commented on ckanext/data_qld/actions.py:
```diff
@@ -91,6 +95,9 @@ def _undictize_datarequest_basic(data_request, data_dict):
     data_request.description = data_dict['description']
     organization = data_dict['organization_id']
     data_request.organization_id = organization if organization else None
+    if tk.h.closing_circumstances_enabled:
```
please make a function as you have this duplicated twice here and on line 335.

MarkCalvert replied:
Refactored into function

Commit:
[DQL2-31]
- Updated config.feature to have correct cucumber 'then' answer
- Refactored duplicate code into method '_undictize_datarequest_closing_circumstances'

Commit:
[DQL-31]
- Updated _undictize_datarequest_closing_circumstances default values to None when empty string
